### ISSUE TITLE
2D augmentation using Compose

### ISSUE THREAD

**Issue by r00tdotexe** [Project-MONAI/tutorials#1445]
Hello,
I am using the MONAI API to build a 2D segmentation pipeline. However, when it comes to creating the Dataset, DataLoader and Transforms for augmentation, I run into a problem.
For the Test Dataset, the transformations are applied to the images and labels differently, hence, the label does not match the area of interest anymore. In a 3D setting, I do not have this problem.

Below you can find the code, the images and labels are stored in separate directories as .npy files.

```
import os
import numpy as np
import matplotlib.pyplot as plt
from glob import glob

from sklearn.model_selection import train_test_split

from monai.transforms import(
    Compose,
    AddChanneld,
    LoadImaged,
    RandAffined,
    RandFlipd,
    RandRotate90d,
    ToTensord,
    # ScaleIntensityRanged,
)
from monai.data import DataLoader, Dataset, CacheDataset, list_data_collate
from monai.utils import set_determinism

path_to_img = r"C:\Users\BKeoh\Data\preprocessed\2D\images"
path_to_label = r"C:\Users\BKeoh\Data\preprocessed\2D\labels"


def prepare(path_to_img, path_to_label):

    images = glob(os.path.join(path_to_img, "*.npy"))
    labels = glob(os.path.join(path_to_label, "*.npy"))

    train_images, valid_images, train_labels, valid_labels = train_test_split(images, 
                                                                            labels, 
                                                                            test_size=0.2, 
                                                                            shuffle=False)

    # After the split, the image and label still fit, I checked this

    train_files = [{'image': image_name, 'label': label_name} for image_name, label_name in zip(train_images, train_labels)]
    valid_files = [{'image': image_name, 'label': label_name} for image_name, label_name in zip(valid_images, valid_labels)]

    # set deterministic training for reproducibility
    set_determinism(seed=0)

    # Define the data transforms and augmentations (random rotation and flips) 
    # TODO: Check bone window using ScaleIntensityRanged()
    train_transforms = Compose(
        [
        LoadImaged(keys=['image', 'label']),
        AddChanneld(keys=['image', 'label']), # Add a channel as 1st element
        RandAffined(
            keys=['image', 'label'],
            mode=('bilinear', 'nearest'),
            prob=0.5,
            rotate_range=(0.174533, 0.174533) # +-10° in radians
            ),
        RandFlipd(keys=["image", "label"], 
                    prob=0.5, 
                    spatial_axis=(0, 1)
                    ),
        RandRotate90d(keys=["image", "label"], 
                    prob=0.5, 
                    max_k=1,
                    spatial_axes=(0, 1)
                    ),
        ToTensord(keys=['image', 'label'])
        ]
    )

    valid_transforms = Compose(
        [
        LoadImaged(keys=['image', 'label']), 
        AddChanneld(keys=['image', 'label']), 
        ToTensord(keys=['image', 'label'])
        ]
    )

    train_ds = Dataset(data=train_files, transform=train_transforms)
    valid_ds = Dataset(data=valid_files, transform=valid_transforms)

    return train_ds, valid_ds

train_ds, valid_ds = prepare(path_to_img, path_to_label)

# Check if augs were applied correctly
n = 1200

plt.imshow(train_ds[n]['image'].squeeze(), cmap='bone')
mask_ = np.ma.masked_where(train_ds[n]['label'].squeeze()==0, train_ds[n]['label'].squeeze())
plt.imshow(mask_, cmap="autumn")

plt.show()
```

**Comment by silver-18** [comment]
I encountered the same issue as you did. Take RandRotate90d as example.
```python
RandRotate90d(keys=["img","seg"], prob=0.5, spatial_axes=[0,1])
``` 
Sometimes the image (also on 2D in my case) has been rotated by 90 degrees, while the labels may have been rotated by 180 degrees.
I wonder if it is really a bug?

**Comment by r00tdotexe** [comment]
Hey,
thanks for your reply. I figured out as well, that its an issue with calling the dataset.
After loading everything into the Dataloader, I extracted a batch and checked again, everything was fine!

Cheers
